### PR TITLE
Header title match

### PR DESCRIPTION
**Pull request by carltonsmith**
### This pull request changes...

- Updates the frontend application so that each page with an H1 header will now use the H1 text and prepend it to the page title

## TO TEST

1. Go to the [TDP dev site](https://tdp-frontend.app.cloud.gov/) and login.
2. Observe the title text on the page.
3. Look at the title in the browser tab.
4. Confirm that they match
5. Complete the form and submit to go to the confirmation
6. Repeat steps 2-4 above

### This pull request is ready to merge when...

- [x] Meets acceptance criteria for the issue 
- [x] Code is reviewed by someone other than the original author
- [x] Code is tested 
- [x] Experience is approved by UX  
- [x] Meets [a11y checklist](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)
- [x] Meets [Raft's Manual QASP Checklist](https://teams.microsoft.com/l/file/B5285301-13CA-46FD-A723-AFFD578C224A?tenantId=d58addea-5053-4a80-8499-ba4d944910df&fileType=docx&objectUrl=https%3A%2F%2Fhhsgov.sharepoint.com%2Fsites%2FTANFDataPortalOFA%2FShared%20Documents%2FGeneral%2FManual%20Testing%20Practices_Issue173.docx&baseUrl=https%3A%2F%2Fhhsgov.sharepoint.com%2Fsites%2FTANFDataPortalOFA&serviceName=teams&threadId=19:f769bbcb029f4f02b55ae7fad90e310d@thread.skype&groupId=41f194a6-c1d3-4680-933e-c8ee7d17e287):lock: (only applicable for PR to main HHS)
- [x] Documentation is updated 
  - OpenAPI  
  - Readme
  - Entity Relationship Diagram (ERD)

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [ ] Each pages site title starts with the h1 content of the page

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: 100% (see https://github.com/raft-tech/TANF-app README coverage dashboard)
  + [ ] Backend coverage: 100% (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Does this PR change any linting or CI settings?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

+ [ ] Did automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [ ] If this PR introduces backend code, is that code documented both inline and overall?
+ [ ] If this PR introduces frontend code, is that code documented both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any security issues?

### CODE REVIEW

**Comment by alexsoble** [comment]
Thanks @carltonsmith! @ttran-hub and @iamjolly, mind if I turn this over to you for an initial round of accessibility review first, since this an a11y-focused PR. If the changes look good from an a11y point of view, that will inform QASP review.

File: tdrs-frontend/src/components/EditProfile/EditProfile.test.js [hunk 1]
```diff
@@ -19,19 +19,6 @@ describe('EditProfile', () => {
   }
   const mockStore = configureStore([thunk])
 
-  it('should have a card with header Request Access', () => {
-    const store = mockStore(initialState)
-    const wrapper = mount(
```
Comment (riatzukiza) [inline tdrs-frontend/src/components/EditProfile/EditProfile.test.js:24]: This test is no longer relavent to only edit profile page. We run this test with private template

File: tdrs-frontend/src/components/PrivateRoute/PrivateRoute.js [hunk 2]
```diff
@@ -36,13 +37,16 @@ function PrivateRoute({ children, history, path }) {
 
   return authenticated ? (
     <Route path={path}>
```
Comment (riatzukiza) [inline tdrs-frontend/src/components/PrivateRoute/PrivateRoute.js:39]: The main heading of the page and the title of the document are set by the private route. The title gets passed in from the router, to the private template, if the user is authorized.

File: tdrs-frontend/src/components/PrivateRoute/PrivateRoute.test.js [hunk 3]
```diff
@@ -12,46 +12,37 @@ import PrivateRoute from '.'
 describe('PrivateRoute.js', () => {
   const mockStore = configureStore([thunk])
 
-  it('does not return children when user is not authenticated', () => {
-    const store = mockStore({ auth: { authenticated: false } })
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter>
-          <PrivateRoute path="/very-secret-route">
+  const createWrapper = (storeOptions) =>
```
Comment (riatzukiza) [inline tdrs-frontend/src/components/PrivateRoute/PrivateRoute.test.js:15]: We had a lot of tests that followed this same pattern, I abstracted to creation of the private router to a utility to make this test read easier

File: tdrs-frontend/src/components/PrivateTemplate/PrivateTemplate.jsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React, { useEffect } from 'react'
+import PropTypes from 'prop-types'
```
Comment (riatzukiza) [inline tdrs-frontend/src/components/PrivateTemplate/PrivateTemplate.jsx:2]: Private template is the main addition from this pull request

File: tdrs-frontend/src/components/Request/Request.jsx [hunk 5]
```diff
@@ -10,11 +10,8 @@ function Request() {
   }
 
   return (
-    <div className="grid-container margin-top-3">
+    <>
```
Comment (riatzukiza) [inline tdrs-frontend/src/components/Request/Request.jsx:13]: Same here as edit profile, the container is being handled by the template

File: tdrs-frontend/src/components/PrivateTemplate/PrivateTemplate.jsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+
+export default function PrivateTemplate({ title, children }) {
+
+    useEffect(() => {
+        document.title = `${title} - TDP - TANF Data Portal`
+    })
+
+    return (
+        <div className="grid-container margin-top-3" >
```
Comment (riatzukiza) [inline tdrs-frontend/src/components/PrivateTemplate/PrivateTemplate.jsx:12]: Same class as the container that use to be repeated on each of the existing pages

File: tdrs-frontend/src/components/Routes/Routes.js [hunk 7]
```diff
@@ -20,10 +20,10 @@ const Routes = () => {
       <Route exact path="/login">
         <LoginCallback />
       </Route>
-      <PrivateRoute exact path="/edit-profile">
+      <PrivateRoute exact title="Request Access" path="/edit-profile">
         <EditProfile />
       </PrivateRoute>
-      <PrivateRoute exact path="/request">
+      <PrivateRoute exact title="Request Submitted" path="/request">
```
Comment (riatzukiza) [inline tdrs-frontend/src/components/Routes/Routes.js:26]: The titles that were previously in the h1 tag of each of these pages is now passed in via title prop to the router.

**Comment by alexsoble** [comment]
I hear that a11y review is going well, in the meantime I'll leave a few quick comments as I scan ahead to the full QASP review!

File: tdrs-frontend/src/axios-instance.js [hunk 8]
```diff
@@ -2,6 +2,8 @@ import axios from 'axios'
 
 // Need a custom instance of axios so we can set the csrf keys on auth_check
 // Work around for csrf cookie issue we encountered in production.
-// It may still be possible to do this with a cookie, and something on the frontend (most likely) is misconfigured.
-// the configuration has alluded us thus far, and this implementation is functionally equivolent to using cookies.
+// It may still be possible to do this with a cookie, and something on the
+// frontend (most likely) is misconfigured. the configuration has alluded
+// us thus far, and this implementation is functionally equivolent to
```
Comment (alexsoble) [inline tdrs-frontend/src/axios-instance.js:7]: Spelling on equivalent

File: tdrs-frontend/src/components/PrivateTemplate/PrivateTemplate.jsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import React, { useEffect } from 'react'
+import PropTypes from 'prop-types'
+
+
+export default function PrivateTemplate({ title, children }) {
```
Comment (alexsoble) [inline tdrs-frontend/src/components/PrivateTemplate/PrivateTemplate.jsx:5]: One of the QASP items refers to documentation, should we add documentation for this component? `PrivateTemplate` alone doesn't tell me what this component is doing or how to use it.

**Comment by iamjolly** [comment]
This PR looks good from the accessibility perspective and is approved from my accessibility-only review.

**Comment by alexsoble** [comment]
# QASP Review 

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [ ] Each pages site title starts with the h1 content of the page.

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested? — Yes, see `PrivateRoute.test.js` and `PrivateTemplate.test.js`
+ Are code coverage minimums met?
  + [x] Frontend coverage: 100%
  + [x] Backend coverage: 100%

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Backend code style checks passing on CircleCI
+ [x] Frontend code style checks passing on CircleCI

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

+ [x] Automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights revealed no errros.
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

+ [x] Code successfully deployed via automated CircleCI process to development on Cloud.gov

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [x] If this PR introduces frontend code, is that code documented both inline and overall? — Yes, see `PrivateTemplate.js`

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] OWASP Scan passes on CircleCI

**Comment by alexsoble** [comment]
@lfrohlich, QASP review items 2 through 7 all pass. How do you feel about item 1, Accepted Features? If you feel the Acceptance Criteria are met via this morning's demo and the development site, I recommend that you accept this work and merge this PR.

**Review by lfrohlich — APPROVED** [review]

**Pull request merged**